test/unit/client/throw-in-type-parser-tests.js

```diff
@@ -0,0 +1,111 @@
+var helper = require(__dirname + "/test-helper");
+var types = require('pg-types')
+
+test('handles throws in type parsers', function() {
+
+  types.setTypeParser('special oid that will throw', function () {
+    throw Object.assign(new Error('Test'), { throwInTypeParser: true });
```

Comment:
It might be best to create the error outside the type parser and check that it’s equal to the emitted error rather than adding `throwInTypeParser`.